theano/scan_module/scan_opt.py

```diff
@@ -2070,6 +2072,7 @@ def map_out(outer_i, inner_o, outer_o, seen):
             new_outer_out_mit_mot.append(outer_omm)
     na.outer_out_mit_mot = new_outer_out_mit_mot
 
+    return dict([("remove", remove)] + zip(node.outputs, na.outer_outputs))
```

Comment:
I think we need to test this a little bit more before merging. It could trigger false user error, as if they get merged, then it would be probably fine.